lib/grandpa/grandpa.go

```diff
@@ -807,16 +808,20 @@ func (s *Service) createJustification(bfc common.Hash, stage Subround) ([]Signed
 		spc  *sync.Map
 		err  error
 		just []SignedVote
+		eqv  map[ed25519.PublicKeyBytes][]*SignedVote
 	)
 
 	switch stage {
 	case prevote:
 		spc = s.prevotes
+		eqv = s.pvEquivocations
 	case precommit:
 		spc = s.precommits
+		eqv = s.pcEquivocations
+	default:
+		return nil, fmt.Errorf("%w: %v", ErrUnsupportedSubround, stage)
```

Comment:
nit use `%s` so it will use `.String` on stage. Not sure if `%v` uses the String method by default, but feel free to enlighten me!

```suggestion
		return nil, fmt.Errorf("%w: %s", ErrUnsupportedSubround, stage)
```